baseplate/frameworks/pyramid/__init__.py

```diff
@@ -325,14 +329,22 @@ def _on_new_request(self, event: pyramid.events.ContextFound) -> None:
             if self.edge_context_factory:
                 request.edge_context = self.edge_context_factory.from_upstream(edge_payload)
 
+        endpoint = ""
+        if request.matched_route:
+            endpoint = (
+                request.matched_route.pattern
+                if request.matched_route.pattern
```

Comment:
_one time_ I ran into an attribute error saying that `pattern` was not available on types.SimpleNamespace. Since then, it's been impossible for me to reproduce which is slightly worrisome to me. I noticed it with the init job in harold. Maybe it was old code that I caught in the middle of a rebuild, but if it wasn't, then this should be changed to a hasattr check.